cmd/vfkit/main.go

```diff
@@ -168,10 +171,17 @@ func runVirtualMachine(vmConfig *config.VirtualMachine, vm *vz.VirtualMachine) e
 			listenStr = " (listening)"
 		}
 		log.Infof("Exposing vsock port %d on %s%s", port, socketURL, listenStr)
-		if err := vf.ExposeVsock(vm, port, socketURL, vsock.Listen); err != nil {
+		if closer, err := vf.ExposeVsock(vm, port, socketURL, vsock.Listen); err != nil {
 			log.Warnf("error exposing vsock port %d: %v", port, err)
+		} else {
+			closes = append(closes, closer)
 		}
```

Comment:
I don't think you need `closes`, this can be
```
if closer, err := vf.ExposeVsock(vm, port, socketURL, vsock.Listen); err != nil {
	log.Warnf("error exposing vsock port %d: %v", port, err)
        continue
}
defer closer.Close()
```